eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/AbstractMutableBiMap.java

```diff
@@ -301,6 +302,23 @@ public V removeKey(K key)
         return this.remove(key);
     }
 
+    @Override
+    public boolean removeAllKeys(Set<? extends K> keys)
+    {
+        Objects.requireNonNull(keys);
+        int previousSize = this.size();
+
+        if (this.size() > keys.size())
+        {
+            keys.forEach(this::removeKey);
+        }
+        else
+        {
+            this.keySet().removeIf(keys::contains);
```

Comment:
It just occurred to me that this is only an optimization if the keys parameter is a hashtable. If it's a tree, it could be slower. It's only guaranteed to be faster if this instance is also a tree. What do you think? The whole thing is such a hypothetical that will never happen in our code.

Reply:
If it’s a Tree we should just iterate over the keys, but what should I do an instance check with? I can check for TreeSet so, the If will get an ‘or’ as well. Any other comments? I would like to get this merged and prepare for 10.0

Reply:
Well I'm really asking for your thoughts here, not suggesting you add an instanceof. If you do want to add an instanceof, maybe NavigableSet makes sense. But maybe it's ok as is since this situation seems so rare. Or maybe we should get rid of the optimization entirely because it's not "worth it". What do you think?

Reply:
I think it is very rare for the keys to be Ordered or Sorted. I am going to add that in the javadocs that it might be slower. I will add the workaround as well in the Javadoc.